simulator/simulator.cpp

```diff
@@ -37,44 +38,48 @@ class SimulatorFactory {
 
     template<typename T, Endian e>
     struct TBuilder : public Builder {
-        TBuilder() = default;
-        std::unique_ptr<Simulator> get_funcsim( bool log) final { return std::make_unique<FuncSim<T>>( e, log); }
-        std::unique_ptr<CycleAccurateSimulator> get_perfsim() final { return std::make_unique<PerfSim<T>>( e); }
+        TBuilder( std::string_view i) { isa = i; }
```

Comment:
This ctor has to be `explicit` now to prevent accidental conversion from `std::string_view`